### ISSUE TITLE
Error installing on windows 7 & 8

### ISSUE THREAD

**Issue by andreofthecape** [proxylocal/proxylocal-gem#11]
When I try to install proxylocal on Windows 7 (also 8) I get this error

C:\Sites\test>gem install proxylocal
Fetching: bert-1.1.6.gem (100%)
Temporarily enhancing PATH to include DevKit...
Building native extensions.  This could take a while...
ERROR:  Error installing proxylocal:
        ERROR: Failed to build gem native extension.

```
    C:/RailsInstaller/Ruby1.9.3/bin/ruby.exe extconf.rb
```

creating Makefile

make
generating decode-i386-mingw32.def
compiling decode.c
decode.c:3:24: fatal error: netinet/in.h: No such file or directory
compilation terminated.
make: **\* [decode.o] Error 1

Gem files will remain installed in C:/RailsInstaller/Ruby1.9.3/lib/ruby/gems/1.9
.1/gems/bert-1.1.6 for inspection.
Results logged to C:/RailsInstaller/Ruby1.9.3/lib/ruby/gems/1.9.1/gems/bert-1.1.
6/ext/bert/c/gem_make.out

**Comment by thekosmix** [comment]
any luck on solving this problem, i'm getting the same

**Comment by thekosmix** [comment]
Found the solution, it's not proxylocal's error, it's bert's error. To install on windows, follow the steps:

``` ruby
git clone https://github.com/mojombo/bert.git
```

it will download bert gem in bert folder.
open file: bert->ext->bert->c->decode.c
replace following header declaration:

``` c
#include <stdint.h>         //linux header file
#include <netinet/in.h>   //linux header file
```

with

``` c
#ifdef _WIN32_        // it it's a windows platform, include windows headers
#include <winsock2.h>     //windows headers
#include <windows.h>     //windows headers
#else                     // else include linux headers
#include <stdint.h>       //linux header file
#include <netinet/in.h>      //linux header file
#endif
```

now build and install bert:

``` ruby
gem build bert.gemspec
gem install bert.-X.X.X.gem
```

now it (bert) will be successfully installed. Now install the proxylocal and run it (in a new cmd):

``` ruby
gem install proxylocal
proxylocal 3000 --host testserver
```

now open url:
http://testserver.t.proxylocal.com/

and it will be displaying your local webserver's content

**Comment by luislavena** [comment]
@thelakshya actually you should be able to check if `_WIN32` is defined and use that to load the right headers, not removal of the other code.

That way, your fix can be incorporated into the gem and everybody benefit, while your current approach just breaks it for anybody else that is not using Windows.

**Comment by thekosmix** [comment]
Agree, I updated my solution, after testing it. Thanks for correcting me!!

**Comment by maurozadu** [comment]
thanks @thelakshya! you saved my life!
@luislavena thanks to you too for correcting the `_WIN32` constant

**Comment by Zaylril** [comment]
Just a quick note in the example you have _WIN32_ it needs to be _WIN32 to work. Just incase others have this issue. Cheers for the fix!

**Comment by seadynamic8** [comment]
@Zaylril - Thanks.  I needed that little adjustment to make it work.

**Comment by daniyal229** [comment]
Great job @thelakshya +1 for the fix
